module-system/sov-modules-stf-template/src/app_template.rs

```diff
@@ -59,92 +59,94 @@ where
         + TxHooks<Context = C>
         + ApplyBlobHooks<Context = C, BlobResult = SequencerOutcome>,
 {
-    fn init_sequencer_and_get_working_set(
+    pub fn new(storage: C::Storage, runtime: RT) -> Self {
+        Self {
+            runtime,
+            current_storage: storage,
+            checkpoint: None,
+            phantom_vm: PhantomData,
+        }
+    }
+
+    pub(crate) fn apply_batch(
         &mut self,
         sequencer: &[u8],
-        batch_data_and_hash: &BatchDataAndHash,
-    ) -> Result<WorkingSet<C::Storage>> {
+        batch: impl Buf,
+    ) -> ApplyBatchResult<BatchReceipt<SequencerOutcome, TxEffect>> {
+        let batch_data_and_hash = BatchDataAndHash::new::<C>(batch);
         debug!(
             "Applying batch from sequencer: 0x{}",
             hex::encode(sequencer)
         );
+
+        // Point A:
+        // Initial batch workspace
```

Comment:
Nit: this comment is a bit confusing. Maybe we could change to `// Initialize batch workspace`?